docs/platform-integration/essentials/geolocation.md

```diff
@@ -1,136 +1,163 @@
 
-The Geolocation API will also prompt the user for permissions when necessary.
+> [!TIP]
+> When necessary, the Geolocation API prompt's the user for permissions.
 
-You can get the last known [location](xref:Microsoft.Maui.Essentials.Location) of the device by calling the `GetLastKnownLocationAsync` method. This is often faster then doing a full query, but can be less accurate and may return `null` if no cached location exists.
+The following code example demonstrates checking for a cached location:
 
 ```csharp
-try
+public async Task GetCachedLocation()
 {
-    var location = await Geolocation.GetLastKnownLocationAsync();
+    try
+    {
+        Location location = await Geolocation.GetLastKnownLocationAsync();
 
-    if (location != null)
+        if (location != null)
+            Console.WriteLine($"Latitude: {location.Latitude}, Longitude: {location.Longitude}, Altitude: {location.Altitude}");
+    }
+    catch (FeatureNotSupportedException fnsEx)
     {
-        Console.WriteLine($"Latitude: {location.Latitude}, Longitude: {location.Longitude}, Altitude: {location.Altitude}");
+        // Handle not supported on device exception
+    }
+    catch (FeatureNotEnabledException fneEx)
+    {
+        // Handle not enabled on device exception
+    }
+    catch (PermissionException pEx)
+    {
+        // Handle permission exception
+    }
+    catch (Exception ex)
+    {
+        // Unable to get location
     }
-}
-catch (FeatureNotSupportedException fnsEx)
-{
-    // Handle not supported on device exception
-}
-catch (FeatureNotEnabledException fneEx)
-{
-    // Handle not enabled on device exception
-}
-catch (PermissionException pEx)
-{
-    // Handle permission exception
-}
-catch (Exception ex)
-{
-    // Unable to get location
 }
 ```
 
-To query the current device's [location](xref:Microsoft.Maui.Essentials.Location) coordinates, the `GetLocationAsync` can be used. It is best to pass in a full `GeolocationRequest` and `CancellationToken` since it may take some time to get the device's location.
+Not all location values may be available, depending on the device. For example, the `Altitude` property might be `null`, have a value of 0, or have a positive value indicating the meters above sea level. Other values that may not be present include `Speed` and `Course`.
+
+## Get the current location
+
+While checking for the [last known location](#get-the-last-known-location) of the device may be quicker, it can be inaccurate. Use the `Geolocation.GetLocationAsync` method to query the device for the current location. You can configure the accuracy and timeout of the query. It's best to use both `GeolocationRequest` and `CancellationToken` parameters, since it may take some time to get the device's location.
```

Comment:
Maybe drop both? I read it as both `GeolocationRequest` params, and both `CancellationToken` params.